src/runner.ts

```diff
@@ -0,0 +1,157 @@
+import { SourceUnit } from "solc-typed-ast";
+import { compile, Result as CompilationResult, unitToSourceCode } from "./compile";
+import { Fuzzer, FuzzingResult } from "./fuzzer";
+import { applyNRandomRewrites, pickAny, Rewrite } from "sol-fuzz";
+import { deepStrictEqual } from "node:assert";
+import logger from "./logging";
+import { writeFile } from "node:fs/promises";
+import path from "node:path";
+import { exists } from "./utils";
+
+function deepEqual(a: any, b: any) {
+    try {
+        deepStrictEqual(a, b);
+    } catch (e) {
+        return false;
+    }
+    return true;
+}
+
+async function saveVariant(variantFileName: string, variant: SourceUnit, version: string) {
+    if (!(await exists(variantFileName))) {
+        await writeFile(
+            variantFileName,
+            // versions[0] is the regular solc version
+            await unitToSourceCode(variant, version)
+        );
+    }
+}
+
+export async function run({
+    seedUnits,
+    rewrites,
+    rewriteDepth,
+    versions,
+    numberOfTests,
+    testCallFunctionName,
+    saveVariants,
+    baseFileName,
+    outputPath
+}: {
+    seedUnits: SourceUnit[];
+    rewrites: Rewrite[];
+    rewriteDepth: number;
+    versions: string[];
+    numberOfTests: number;
+    testCallFunctionName: string;
+    saveVariants: boolean;
+    baseFileName: string;
+    outputPath: string;
+}): Promise<{
+    compilationFailures: Array<[string, CompilationResult]>;
+    fuzzingFailures: Array<[string, string, FuzzingResult]>;
+}> {
+    const compilationFailures: Array<[string, CompilationResult]> = [];
+    const fuzzingFailures: Array<[string, string, FuzzingResult]> = [];
+
+    for (let i = 0; i < numberOfTests; i++) {
```

Comment:
The logic in this function is kinda hard to follow. A potentially simpler logic could be to generate a vector of `Array<CompilationResult | FuzzingResult>` with the results for each version for a given test file. If any two entries in the vector are different, add the whole vector as filed output.

Consequentially in the `results.json` we would have a 2d array with one row for each test case where some results differ for some versions. In that row we would have the compilation/fuzzing results for all version for that test file, not just the differing ones.

Reply:
Fixed in [93d139d](https://github.com/ConsenSysDiligence/solc-fuzz/pull/3/commits/93d139d70ab85ce749f0f582f9f3e8ef16784f5c)

Basically we changed the output format to be:

For any test, for which two of the versions differ, just output the results for *all* the versions.